packages/salesforcedx-vscode-core/src/commands/forcePackageInstall.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import {
+  CancelResponse,
+  ContinueResponse,
+  ParametersGatherer
+} from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import { Observable } from 'rxjs/Observable';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { nls } from '../messages';
+import { notificationService, ProgressNotification } from '../notifications';
+import { taskViewService } from '../statuses';
+import { getRootWorkspacePath } from '../util';
+import {
+  CompositeParametersGatherer,
+  EmptyPreChecker,
+  SfdxCommandlet,
+  SfdxCommandletExecutor
+} from './util';
+
+type forcePackageInstallOptions = {
+  packageId: string;
+  installationKey: string;
+};
+
+export class ForcePackageInstallExecutor extends SfdxCommandletExecutor<
+  PackageIdAndInstallationKey
+> {
+  private readonly options: forcePackageInstallOptions;
+
+  public constructor(options = { packageId: '', installationKey: '' }) {
+    super();
+    this.options = options;
+  }
+
+  public build(data: PackageIdAndInstallationKey): Command {
+    const builder = new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_package_install_text'))
+      .withArg('force:package:install')
+      .withFlag('--package', data.packageId)
+      .withLogName('force_package_install');
+
+    if (data.installationKey) {
+      builder.withFlag('--installationkey', data.installationKey);
+    }
+
+    return builder.build();
+  }
+
+  public execute(
+    response: ContinueResponse<PackageIdAndInstallationKey>
+  ): void {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath()
```

Comment:
We may want to add this piece as well...
env: { SFDX_JSON_TO_STDOUT: 'true' }

Reply:
Done! Thanks for pointing this out